### PR TITLE
Set player can fly fall damage

### PR DESCRIPTION
**Pull request by TreyRuffy**
I have added a method in order to allow fall damage, even with allowFlight enabled. This is mostly for my own double jump plugin but can be used anywhere.

I am happy to make changes to this and I have left allow edits by maintainers. Feel free to ask any questions about this.

I originally made a PR to Paper (PaperMC/Paper#5357) but it looks like it isn't going anywhere.

### CODE REVIEW

**Comment by YouHaveTrouble** [comment]
I'm actually all for this, never liked how people with /fly just didn't take fall damage at all.

**Comment by katekerllenevich** [comment]
the method names seem kinda lengthy

**Comment by TreyRuffy** [comment]
> the method names seem kinda lengthy

I agree but I also can't think of anything shorter

File: patches/api/0042-Can-Fly-Fall-Damage.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      */
+     public void setAllowFlight(boolean flight);
+ 
++    // Purpur start - Add method to enable fall damage if you have the allowFlight attribute
++    /**
++     * Allows you to enable fall damage while {@link #getAllowFlight()} is {@code true}
++     *
++     * @param fallDamage Enables fall damage when {@link #getAllowFlight()} is true
++     */
++    public void setEnableFallDamageWhileAllowFlight(boolean fallDamage);
```
Comment (BillyGalbreath) [inline patches/api/0042-Can-Fly-Fall-Damage.patch]:
```suggestion
+    public void setFlyingFallDamage(boolean flyingFallDamage);
```

File: patches/api/0042-Can-Fly-Fall-Damage.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
++
++    /**
++     * Allows you get if fall damage is enabled while {@link #getAllowFlight()} is {@code true}
++     *
++     * @return True if fall damage is enabled when {@link #getAllowFlight()} is true
++     */
++    public boolean getEnableFallDamageWhileAllowFlight();
```
Comment (BillyGalbreath) [inline patches/api/0042-Can-Fly-Fall-Damage.patch]:
```suggestion
+    public boolean hasFlyingFallDamage();
```

**Review by BillyGalbreath — COMMENTED** [review]
Looks good to me. Here's some shorter method name suggestions

**Pull request merged**